File: InterviewAI-AIMicroservice/services_types/system_design.py
from pydantic import BaseModel, Field
from typing import Union

class SystemDesignQuestion(BaseModel):
    question: str
    answer: str
    designed_system_image_url: Union[str, None] = None

class Grade(BaseModel):
    grade: int = Field(description="This is the grade for the user that the user will get based on their performance in the system design round. This should be between 0 and 100. THis field is required.")
    feedback: str = Field(description="This is the detailed feedback of the user's performance in the system design round. This field is required.")

File: InterviewAI-Backend/src/types/SystemDesign.ts
export interface SystemDesignQuestion {
    question: string;
    difficulty: string;
}

export interface SystemDesignSubmission {
    question: string;
    answer: string;
    designed_system_image_base64: string;
}

export interface SystemDesignSubmissionAIRequest {
    question: string;
    answer: string;
    designed_system_image_url: string;
}

export interface SystemDesignSubmissionAIResponse {
    grade: number;
    feedback: string;
}